source/_posts/2016-09-14-what_the_tag_series_ep1_document_object.markdown

```diff
@@ -0,0 +1,123 @@
+
+`document.lastModified`
+
+It lets you know about when the document was modified for the last time, this info is retrieved from the server response, in case this is not available the current time is returned in the local time zone.
+
+`document.readyState`
+
+This readonly attribute is a very important and useful one, it holds the current state of the document, it can be in three different states:
+
+**loading**
+When the document has not finished loading (the response from the server is not yet completed).
+
+**interactive**
+In case the first load is done and the parsing process is completed but there are some resources that need to be loaded, for example images, styles, javascript, etc.
+
+**complete**
+When all related resources are loaded. Once this state is achieved the `load` event will be fired.
+
+Every time the value of this attribute changes the `readystatechange` event is fired. This event is equivalent to the more known ones `load` and `DOMContentLoaded`, next is an example, that I borrowed from the docs of [Mozilla](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState) (if you need more cool examples just visit them!), this simply listens to the event and does some application initialization:
+
+```js
+// alternative to DOMContentLoaded event
+document.onreadystatechange = function () {
+  if (document.readyState === "interactive") {
+    initApplication();
+  }
+}
+```
+
+## DOM tree accessors
+
+This accessors let you interact directly with the document, by setting the values of this properties you can modify the document and change it to your will, but remember, as uncle Ben once said:
+> With great power comes great responsibility.
+
+`document.head`
+
+This returns the head element, and it is a readonly attribute. The head element is where the metadata of the document is, for example the title, styles and scripts.
+
+`document.title`
+
+This attribute is a very simple one, it lets you get the title of the document or set it. And yes, the title is the thing is written in the browser tab.
+
+`document.body`
+
+Here is where probably most of your website's HTML is, here you can access the code directly and change it. I invite you to try to break this website, investigate about `innerHTML`, (in case you succeed just refresh the site).
+
+`document.images`
+
+Using this you can get a list of all images in the document, but it is read only, it is just an easy way to access this kind of elements, you can affect this indirectly by modifying the document and adding more images. There are more than this accessor in the list, take a look at the [definition of the images](https://html.spec.whatwg.org/multipage/dom.html#dom-document-images).
+
```

Comment:
Not sure what this line means: `There are more than this accessor in the list`, is it that there are similar properties like `embeds`, `plugins`, `links`? If so, I would say something more like `There are other accessors similar to this one, take a look...`